eclipse/src/saros/editor/EditorPool.java

```diff
@@ -310,4 +314,36 @@ public void setElementStateListenerEnabled(final boolean enabled) {
   public void setDocumentListenerEnabled(final boolean enabled) {
     documentListener.setEnabled(enabled);
   }
+
+  private static void findAndLogDocumentProviderIssues(final IEditorPart editorPart) {
+
+    final IDocumentProvider defaultDocumentProvider =
+        EditorAPI.getDocumentProvider(editorPart.getEditorInput());
+
+    if (!(defaultDocumentProvider instanceof TextFileDocumentProvider)) {
+      LOG.warn(
+          "The default document provider "
+              + defaultDocumentProvider
+              + " for editor with title '"
+              + editorPart.getTitle()
+              + "' might not support shared access. It is likely that the editor content is not properly synchronized!");
+
+      return;
+    }
+
+    final ITextEditor textEditor = editorPart.getAdapter(ITextEditor.class);
+
+    if (textEditor == null) return;
+
+    final IDocumentProvider editorDocumentProvider = textEditor.getDocumentProvider();
+
+    if (!(defaulDocumentProvider instanceof TextFileDocumentProvider)) {
```

Comment:
Seems like you missed one.
```suggestion
    if (!(defaultDocumentProvider instanceof TextFileDocumentProvider)) {
```

Reply:
In theory yes, but looking at it it should be the editorDocumentProvider.

Reply:
Makes sense^^